parlai/agents/drqa/drqa.py

```diff
@@ -83,13 +99,21 @@ def add_to_dict(self, tokens):
 # ------------------------------------------------------------------------------
 
 
-class DocReaderAgent(Agent):
+class DrqaAgent(Agent):
 
     @staticmethod
     def add_cmdline_args(argparser):
         config.add_cmdline_args(argparser)
+        SimpleDictionaryAgent.add_cmdline_args(argparser)
 
-    def __init__(self, opt, shared=None, word_dict=None):
+    @staticmethod
+    def dictionary_class():
+        return "parlai.agents.drqa.drqa:SimpleDictionaryAgent"
+
+    def __init__(self, opt, shared=None):
+        # Load dict.
+        if not shared:
+            word_dict = SimpleDictionaryAgent(opt)
```

Comment:
should this not use the function above to choose which dict class to use? if the other function returned a class then it would be easy, but the string does make it harder...